### ISSUE TITLE
modifications are not saved for some shared folders

### ISSUE THREAD

**Issue by jsuhm** [ONLYOFFICE/onlyoffice-nextcloud#156]
We are using 
- ONLYOFFICE Document Server: Version 5.4.0.21
- Nextcloud: Version  16.0.5-0 with ONLYOFFICE app v3.0.2

In some cases, modifications made in OnlyOffice are not saved to Nextcloud.

### Steps :
(all made inside a web browser)

1. An admin creates a group TESTGROUP with a user A
2. An admin creates a groupfolder "GROUPFOLDER" shared with TESTGROUP
3. User A create a subfolder "SUBFOLDER" and a file "test.docx" inside
4. User A create a Circle "A_Circle" and add user B
5. User A share "SUBFOLDER" with the circle "A_Circle"
6. User B edit the file "test.docx" with OnlyOffice, make some modifications, save and close the tab

### Results :
In A home:
```
    L[ GROUPFOLDER ] --> shared with the group TESTGROUP (A is member, but not B)
        L[ SUBFOLDER ] --> Created by A and shared with the circle "A_Circle", B is member
           L test.docx --> Created by A with OnlyOffice
```
In B home:
```
    [ SUBFOLDER ] --> Created by A and shared with the circle "A_Circle", B is member
           L test.docx --> B edit this file with OnlyOffice and want to save it in NextCloud
```

### Issue : 
Modifications made by B on file test.docx are not saved in NextCloud : if he download the file, his modifications are not present
Note : modifications are visible if the file is edited once again in OnlyOffice (from a OnlyOffice's cache I suppose)

### Expected :
modifications made by B are saved in NextCloud

**Comment by HelenBelova17** [comment]
Same scenario without circles (using groups instead):

Steps (all made inside a web browser):
An admin creates a group TESTGROUP with a user A and TESTGROUP2 with user B
An admin creates a groupfolder "GROUPFOLDER" shared with TESTGROUP
User A create a subfolder "SUBFOLDER" and a file "test.docx" inside
User A share "SUBFOLDER" with TESTGROUP2 (or with user B directly)
User B edit the file "test.docx" with OnlyOffice, make some modifications, save and close the tab

Result : Modifications made by B on file test.docx are not saved in NextCloud.

**Comment by jsuhm** [comment]
The issue seems to come from re-sharing to a user / group / circle who don't have access to the parent groupfolder.
Even if it works without Groupfolders, Groupfolders are mandatory for us
Maybe we should prevent the re-sharing ... and find another method for cross-team work.

**Comment by jlord87** [comment]
I'm facing the same problem of @HelenBelova17.

I'm using onlyoffice-documentserver 5.4.2 on Ubuntu 18.04.2.
Nextcloud Server 16.0.6 on Ubuntu 18.04.2
Onlyoffice app 3.0.2

If you need any further information for debugging or a test user in my environment just ask.

Thank you

**Comment by LinneyS** [comment]
Hello to all
please check version 4.0.0

**Comment by jlord87** [comment]
Awesome @LinneyS, you solved it!
I owe you a beer, write me when you'll be in Italy next time.

Cheers

**Comment by jsuhm** [comment]
> Hello to all
> please check version 4.0.0

Hello LinneyS

Unfortunately I can't test it on my server for the moment.
Do you mean you solved it with OnlyOffice app 4.0 and it works now with groupfolders ?

**Comment by HelenBelova17** [comment]
Hello, @**jsuhm**, I can confirm issue solved in OnlyOffice app 4.0.
